docs/content/concepts/assets/asset-checks.mdx

```diff
@@ -0,0 +1,141 @@
+```
+
+[`@asset_check`](/\_apidocs/asset-checks#dagster.asset_check) is a decorator for a function that returns a [`AssetCheckResult`](/\_apidocs/asset-checks#dagster.AssetCheckResult) object.
+
+Next we'll use a more realistic example. The following code defines an asset named `orders` and an asset check named `orders_id_is_unique`. When executed, the check verifies a property of the `orders` asset: that all the values in its primary key column are unique.
+
+```python file=/concepts/assets/asset_checks/orders_check.py
+import pandas as pd
+
+from dagster import AssetCheckResult, Definitions, asset, asset_check
+
+
+@asset
+def orders():
+    orders_df = pd.DataFrame({"order_id": [1, 2], "item_id": [432, 878]})
+    orders_df.to_csv("orders.csv")
+
+
+@asset_check(asset=orders, description="ensure there are no null order_ids")
+def orders_id_has_no_nulls(context):
+    orders_df = pd.read_csv("orders.csv")
+    num_null_order_ids = orders_df["order_id"].isna().sum()
+    return AssetCheckResult(
+        success=(num_null_order_ids == 0),
+        metadata={"num_null_order_ids": num_null_order_ids},
+    )
+
+
+defs = Definitions(
+    assets=[orders],
+    asset_checks=[orders_id_has_no_nulls],
+)
+```
+
+The `orders_id_is_unique` check runs in its own op. That means that, if you launch a run that materializes the `orders` asset and also executes the `orders_id_is_unique` check, and you’re using the <PyObject module="dagster" object="multiprocess_executor" />, the check will execute in a separate process from the process that materializes the asset.
+
+### Checks that execute in the same op that materializes the asset
+
+Sometimes, it makes the most sense for a single function to both materialize an asset and then execute a check on it.
+
+When defining an asset using the [`@asset`](/\_apidocs/assets#dagster.asset) or [`@multi_asset`](/\_apidocs/assets#dagster.multi_asset) decorators, you can provide values for the `check_specs` argument. Each provided [`AssetCheckSpec`](/\_apidocs/asset-checks#dagster.AssetCheckSpec) declares a check that the decorated function should yield an [`AssetCheckResult`](/\_apidocs/asset-checks#dagster.AssetCheckSpec) for.
+
+```python file=/concepts/assets/asset_checks/asset_with_check.py
+import pandas as pd
+
+from dagster import AssetCheckResult, AssetCheckSpec, Output, asset
+
+
+@asset(check_specs=[AssetCheckSpec("orders_id_has_no_nulls")])
+def orders(context):
+    orders_df = pd.DataFrame({"order_id": [1, 2], "item_id": [432, 878]})
+
+    # save the output and indicate that it's been saved
+    orders_df.to_csv("orders")
+    yield Output(value=None)
+
+    # check it
+    num_null_order_ids = orders_df["order_id"].isna().sum()
+    yield AssetCheckResult(
+        success=(num_null_order_ids == 0),
+        metadata={"num_null_order_ids": num_null_order_ids},
+    )
+```
+
+## Executing asset checks
+
+### Building jobs that include asset checks
```

Comment:
I think this needs a first sentence that provides more context.  E.g. "You can define a job that materializes assets and also executes their checks"